src/frontend/js/widgets/Dashboard/components/DashboardSidebar/components/ContractNavLink/index.spec.tsx

```diff
@@ -0,0 +1,245 @@
+        contractQueryParams,
+        { sort: false },
+      )}`,
+      {
+        count: 1,
+        next: null,
+        previous: null,
+        results: [ContractFactory({ abilities: { [ContractActions.SIGN]: false } }).one()],
+      },
+    );
+
+    render(
+      <Wrapper>
+        <ContractNavLink
+          link={{
+            to: '/dummy/url/',
+            label: 'My contract navigation link',
+          }}
+          organizationId={organizationId}
+        />
+      </Wrapper>,
+    );
+
+    expect(screen.getByRole('link', { name: 'My contract navigation link' })).toBeInTheDocument();
+    expect(screen.queryByTestId('badge')).not.toBeInTheDocument();
+  });
+
+  describe('without sign ability', () => {
+    const contractAbilities = { [ContractActions.SIGN]: false };
+    it.each([
+      {
+        organizationId: faker.string.uuid(),
+        courseProductRelationId: undefined,
+      },
+      {
+        organizationId: faker.string.uuid(),
+        courseProductRelationId: faker.string.uuid(),
+      },
+      {
+        organizationId: undefined,
+        courseProductRelationId: faker.string.uuid(),
+      },
+      {
+        organizationId: undefined,
+        courseProductRelationId: undefined,
+      },
+    ])(
+      'should never render Badge for organizationId: $organizationId and courseProductId: $courseProductRelationId',
+      async ({ organizationId, courseProductRelationId }) => {
+        let contractQueryParams: ContractResourceQuery = {
+          signature_state: ContractState.LEARNER_SIGNED,
+          page: 1,
+          page_size: PER_PAGE.teacherContractList,
+        };
+        if (courseProductRelationId) {
+          contractQueryParams = {
+            course_product_relation_id: courseProductRelationId,
+            ...contractQueryParams,
+          };
+        }
+
+        fetchMock.get(
+          `https://joanie.endpoint/api/v1.0/organizations/${organizationId}/contracts/?${queryString.stringify(
+            contractQueryParams,
+            { sort: false },
+          )}`,
+          {
+            count: 1,
+            next: null,
+            previous: null,
+            results: [ContractFactory({ abilities: contractAbilities }).one()],
+          },
+        );
+        render(
+          <Wrapper>
+            <ContractNavLink
+              link={{
+                to: '/dummy/url/',
+                label: 'My contract navigation link',
+              }}
+              organizationId={organizationId}
+              courseProductRelationId={courseProductRelationId}
+            />
+          </Wrapper>,
+        );
+
+        expect(
+          screen.getByRole('link', { name: 'My contract navigation link' }),
+        ).toBeInTheDocument();
+        expect(screen.queryByTestId('badge')).not.toBeInTheDocument();
+      },
+    );
+  });
+
+  describe('with sign ability', () => {
```

Comment:
I think a test is missing to check that even if the user has ability to sign, if there is no contract to sign, no badge should be displayed and the link should be redirect to the list of signed contracts.

Reply:
I've added some cases in my `each` test: 
```shell
with sign ability
     # new cases
     ✓ should render Badge (count: undefined) for nb contracts to sign: 0, organizationId: 694f6af0-d293-44c8-8a20-8acc9250fb79 and courseProductId: undefined (16 ms)
      ✓ should render Badge (count: undefined) for nb contracts to sign: 0, organizationId: 0a647ead-d3ce-47db-bf29-574158e8aa55 and courseProductId: 2d495605-12d1-4851-a1a6-6d853d4c0e99 (17 ms)
      ✓ should render Badge (count: undefined) for nb contracts to sign: 0, organizationId: undefined and courseProductId: 53c50175-6b1a-4e59-ad52-0f5ea19be200 (14 ms)
      ✓ should render Badge (count: undefined) for nb contracts to sign: 0, organizationId: undefined and courseProductId: undefined (11 ms)

```